scripts/quests/sandoria/Unexpected_Treasure.lua

```diff
@@ -0,0 +1,114 @@
+-----------------------------------
+-- Unexpected Treasure
+-----------------------------------
+-- Log ID: 0, Quest ID: 70
+-- Morunaude: !gotoid 17723523
+-- Calovour: !gotoid 17723524
+-----------------------------------
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+
+local quest = Quest:new(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.UNEXPECTED_TREASURE)
+
+quest.reward =
+{
+    fame = 30,
+    fameArea = xi.quest.fame_area.SANDORIA,
+    gil = 12000,
+
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getFameLevel(xi.quest.fame_area.SANDORIA) >= 4 and
+                player:hasKeyItem(xi.ki.SMALL_TEACUP)
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Morunaude'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(635, 0, xi.ki.SMALL_TEACUP, xi.items.CUPBOARD)
+                end,
+            },
```

Comment:
['Morunaude'] = quest:progressEvent(635, 0, xi.ki.SMALL_TEACUP, xi.items.CUPBOARD),

can shrink that down to just that above instead of having the onTrigger.

Reply:
Updated